tests/integration/targets/metal_plans/tasks/main.yml

```
@@ -0,0 +1,76 @@
+- name: metal_plan_info
+  module_defaults:
+    equinix.cloud.metal_plans_info:
+      metal_api_token: '{{ metal_api_token }}'
+    equinix.cloud.metal_project:
+      metal_api_token: '{{ metal_api_token }}'
+    equinix.cloud.metal_project_info:
+      metal_api_token: '{{ metal_api_token }}'
+      metal_ua_prefix: '{{ metal_ua_prefix }}'
+  block:
+    - set_fact:
+        test_resource_name_prefix: 'ansible-integration-test-plan'
+
+    - set_fact:
+        unique_id: "{{ lookup('community.general.random_string', upper=false, numbers=false, special=false) }}"
+
+    - set_fact:
+        test_prefix: "{{ test_resource_name_prefix }}-{{ unique_id }}"
+
+    - name: create project for test
+      equinix.cloud.metal_project:
+        name: "{{ test_prefix }}-project"
+      register: project
+
+    - assert:
+        that:
+        - project.name == '{{ test_prefix }}-project'
+
+    - name: list metal plans info by slug
+      equinix.cloud.metal_plans_info:
+        slug: c3.medium
+
+    - name: list metal plans info by project
+      equinix.cloud.metal_plans_info:
+        project_id: "{{ project.id }}"
+
+    - name: list metal plans info by organization
+      equinix.cloud.metal_plans_info:
+        organization_id: "{{ project.organization_id }}"
+
+    - name: list all metal plans
+      equinix.cloud.metal_plans_info:
+        organization_id: "{{ project.organization_id }}"
+
+    - name: list metal plans info by slug
+      equinix.cloud.metal_plans_info:
+        categories: [ 'storage' ]
+
+    - name: list metal plans info by type
+      equinix.cloud.metal_plans_info:
+        type: spot_market
```

I think we have to have valid type here. 

```suggestion
        type: standard
```